File: client/bcephal-client-model/Projects/ImportProjectData.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Projects
{
   public class ImportProjectData
    {
        public string FilePath { get; set; }

        public string NewProjectName { get; set; }

        public bool OverwriteExistingProject { get; set; }

        public bool RenameExistingProject { get; set; }

        public bool RenameProjectToImport { get; set; }

        public long ClientId { get; set; }
        public long ProfileId { get; set; }

        public string Locale { get; set; }

        [JsonIgnore]
        public ImportProjectDecision Decision { get; set; }

        public ImportProjectData()
        {
            this.OverwriteExistingProject = false;
            this.RenameExistingProject = false;
            this.RenameProjectToImport = false;
        }
    }

    public enum ImportProjectDecision
    {
        OverwriteExistingProject,
        RenameExistingProject,
        RenameProjectToImport
    }
}

File: server/bcephal-base/src/main/java/com/moriset/bcephal/project/archive/ImportProjectData.java
package com.moriset.bcephal.project.archive;

import java.util.Locale;

import lombok.Data;

@Data
public class ImportProjectData {
	private String filePath;

	private String newProjectName;

	private boolean overwriteExistingProject;

	private boolean renameExistingProject;

	private boolean renameProjectToImport;

	private Long clientId;
	private Long profileId;

	private Locale locale;

	public ImportProjectData() {
		this.overwriteExistingProject = false;
		this.renameExistingProject = false;
		this.renameProjectToImport = false;
	}
}
